FocusManager: Allow preventing to scroll when focusing an element

gabrielstalder opened the pull request:
Change to support _preventScroll_ when focusing elements. The second commit uses this feature to improve the TileGrid.

gabrielstalder committed:
FocusManager: Allow preventing to scroll when focusing an element

When focusing an element, some browsers scroll vertically by default.
This may be prevented by the preventScroll parameter for the native
focus call. The FocusManager now supports this option.

300077

cguglielmo commented on eclipse-scout-core/src/focus/FocusManager.js:
```diff
@@ -251,26 +251,43 @@ export default class FocusManager {
   /**
    * Requests the focus for the given element, but only if being a valid focus location.
    *
+   * @param {HTMLElement, $} [element]
```
Please use | instead of , , as it is the official syntax: https://jsdoc.app/tags-param.html?msclkid=1495682dcd2911eca72dcc8e2122179a#multiple-types-and-repeatable-parameters

gabrielstalder committed:
TileGrid: Prevent scrolling on focus with preventScroll option

Focusing a new tile triggers a scroll event. To avoid this, the focus
is now requested with the preventScroll option.

300077
